atc/exec/set_pipeline_step_test.go

```diff
@@ -241,6 +257,54 @@ jobs:
 			})
 		})
 
+		Context("when pipeline file exists but has task syntax", func() {
```

Comment:
For the tests, these won't be valid once we update the `configvalidate` package and they don't express the desired UX. The following is weird but it is essentially the scenario that your team ran into (kinda?).

There's a context `when specified pipeline not found`, and under that could be another context that says `when no diff is found`. That'll push us down the codepath to verify that we don't call `pipeline.SetParentIDs()` when no pipeline was found.
```go
Context("when no diff is found", func() {
  ...
}

Reply:
@taylorsilva Thanks! I _believe_ I've modified the tests as you're describing. Please let me know if I've misinterpreted.